src/snapred/meta/mantid/WorkspaceNameGenerator.py

```diff
@@ -83,17 +82,34 @@ def formatRunNumber(runNumber: str):
         return str(runNumber).zfill(6)
 
     @staticmethod
-    def formatVersion(version: Optional[Version], use_v_prefix: vPrefix = vPrefix.WORKSPACE):
+    def formatVersion(version: Optional[int], use_v_prefix: vPrefix = vPrefix.WORKSPACE):
+        # handle two special cases of unassigned or default version
+        # in these cases, change the value to a user-specified string
         if version is None:
-            return ""
-        if not version == "*":
+            version = ""
+        elif version == VERSION_DEFAULT:
+            version = VERSION_DEFAULT_NAME
+        elif str(version).isdigit():
+            version = int(version)
+        else:
+            version = ""
```

Comment:
this sort of logic feels like it should live in `Versioning` as a static method, then utilized here.